diff -uNr bzip2-1.0.1/bzip2.c bzip2-1.0.1.good/bzip2.c
--- bzip2-1.0.1/bzip2.c	2000-04-25 23:34:06.000000000 +0200
+++ bzip2-1.0.1.good/bzip2.c	2003-08-27 16:20:48.932133328 +0200
@@ -211,7 +211,7 @@
 #if BZ_LCCWIN32
 #   include <io.h>
 #   include <fcntl.h>
-#   include <sys\stat.h>
+#   include <sys/stat.h>
 
 #   define NORETURN       /**/
 #   define PATH_SEP       '\\'
diff -uNr bzip2-1.0.1/libbz2.def bzip2-1.0.1.good/libbz2.def
--- bzip2-1.0.1/libbz2.def	2000-05-16 23:35:52.000000000 +0200
+++ bzip2-1.0.1.good/libbz2.def	2003-08-27 16:20:03.627020752 +0200
@@ -1,27 +1,27 @@
 LIBRARY			LIBBZ2
 DESCRIPTION		"libbzip2: library for data compression"
 EXPORTS
-	BZ2_bzCompressInit
-	BZ2_bzCompress
-	BZ2_bzCompressEnd
-	BZ2_bzDecompressInit
-	BZ2_bzDecompress
-	BZ2_bzDecompressEnd
-	BZ2_bzReadOpen
-	BZ2_bzReadClose
-	BZ2_bzReadGetUnused
-	BZ2_bzRead
-	BZ2_bzWriteOpen
-	BZ2_bzWrite
-	BZ2_bzWriteClose
-	BZ2_bzWriteClose64
-	BZ2_bzBuffToBuffCompress
-	BZ2_bzBuffToBuffDecompress
-	BZ2_bzlibVersion
-	BZ2_bzopen
-	BZ2_bzdopen
-	BZ2_bzread
-	BZ2_bzwrite
-	BZ2_bzflush
-	BZ2_bzclose
-	BZ2_bzerror
+	BZ2_bzCompressInit@16
+	BZ2_bzCompress@8
+	BZ2_bzCompressEnd@4
+	BZ2_bzDecompressInit@12
+	BZ2_bzDecompress@4
+	BZ2_bzDecompressEnd@4
+	BZ2_bzReadOpen@24
+	BZ2_bzReadClose@8
+	BZ2_bzReadGetUnused@16
+	BZ2_bzRead@16
+	BZ2_bzWriteOpen@20
+	BZ2_bzWrite@16
+	BZ2_bzWriteClose@20
+	BZ2_bzWriteClose64@28
+	BZ2_bzBuffToBuffCompress@28
+	BZ2_bzBuffToBuffDecompress@24
+	BZ2_bzlibVersion@0
+	BZ2_bzopen@8
+	BZ2_bzdopen@8
+	BZ2_bzread@12
+	BZ2_bzwrite@12
+	BZ2_bzflush@4
+	BZ2_bzclose@4
+	BZ2_bzerror@8
--- bzip2-1.0.2/Makefile.orig	2002-01-26 00:34:53.000000000 +0100
+++ bzip2-1.0.2/Makefile	2003-12-01 23:22:57.502761240 +0100
@@ -23,7 +23,7 @@
       decompress.o \
       bzlib.o
 
-all: libbz2.a bzip2 bzip2recover test
+all: libbz2.a bzip2 bzip2recover
 
 bzip2: libbz2.a bzip2.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2
@@ -33,12 +33,7 @@
 
 libbz2.a: $(OBJS)
 	rm -f libbz2.a
-	$(AR) cq libbz2.a $(OBJS)
-	@if ( test -f $(RANLIB) -o -f /usr/bin/ranlib -o \
-		-f /bin/ranlib -o -f /usr/ccs/bin/ranlib ) ; then \
-		echo $(RANLIB) libbz2.a ; \
-		$(RANLIB) libbz2.a ; \
-	fi
+	i386-mingw32-dllwrap --def libbz2.def --implib libbz2.a -o libbz2.dll $(OBJS)
 
 check: test
 test: bzip2
