diff -uNr bzip2-1.0.1/libbz2.def bzip2-1.0.1.good/libbz2.def
--- bzip2-1.0.1/libbz2.def	2000-05-16 23:35:52.000000000 +0200
+++ bzip2-1.0.1.good/libbz2.def	2003-08-27 16:20:03.627020752 +0200
@@ -1,27 +1,27 @@
 LIBRARY			LIBBZ2
 DESCRIPTION		"libbzip2: library for data compression"
 EXPORTS
-	BZ2_bzCompressInit
-	BZ2_bzCompress
-	BZ2_bzCompressEnd
-	BZ2_bzDecompressInit
-	BZ2_bzDecompress
-	BZ2_bzDecompressEnd
-	BZ2_bzReadOpen
-	BZ2_bzReadClose
-	BZ2_bzReadGetUnused
-	BZ2_bzRead
-	BZ2_bzWriteOpen
-	BZ2_bzWrite
-	BZ2_bzWriteClose
-	BZ2_bzWriteClose64
-	BZ2_bzBuffToBuffCompress
-	BZ2_bzBuffToBuffDecompress
-	BZ2_bzlibVersion
-	BZ2_bzopen
-	BZ2_bzdopen
-	BZ2_bzread
-	BZ2_bzwrite
-	BZ2_bzflush
-	BZ2_bzclose
-	BZ2_bzerror
+	BZ2_bzCompressInit@16
+	BZ2_bzCompress@8
+	BZ2_bzCompressEnd@4
+	BZ2_bzDecompressInit@12
+	BZ2_bzDecompress@4
+	BZ2_bzDecompressEnd@4
+	BZ2_bzReadOpen@24
+	BZ2_bzReadClose@8
+	BZ2_bzReadGetUnused@16
+	BZ2_bzRead@16
+	BZ2_bzWriteOpen@20
+	BZ2_bzWrite@16
+	BZ2_bzWriteClose@20
+	BZ2_bzWriteClose64@28
+	BZ2_bzBuffToBuffCompress@28
+	BZ2_bzBuffToBuffDecompress@24
+	BZ2_bzlibVersion@0
+	BZ2_bzopen@8
+	BZ2_bzdopen@8
+	BZ2_bzread@12
+	BZ2_bzwrite@12
+	BZ2_bzflush@4
+	BZ2_bzclose@4
+	BZ2_bzerror@8
--- bzip2-1.0.6/Makefile.orig	2010-09-11 00:46:02.000000000 +0200
+++ bzip2-1.0.6/Makefile	2013-06-23 09:39:59.263591821 +0200
@@ -35,7 +35,7 @@
       decompress.o \
       bzlib.o
 
-all: libbz2.a bzip2 bzip2recover test
+all: libbz2.a bzip2 bzip2recover
 
 bzip2: libbz2.a bzip2.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2
@@ -46,11 +46,8 @@
 libbz2.a: $(OBJS)
 	rm -f libbz2.a
 	$(AR) cq libbz2.a $(OBJS)
-	@if ( test -f $(RANLIB) -o -f /usr/bin/ranlib -o \
-		-f /bin/ranlib -o -f /usr/ccs/bin/ranlib ) ; then \
-		echo $(RANLIB) libbz2.a ; \
-		$(RANLIB) libbz2.a ; \
-	fi
+	$(RANLIB) libbz2.a
+	$(CC) -shared $(LDFLAGS) -Wl,--enable-auto-image-base -Wl,--add-stdcall-alias -Wl,--out-implib,libbz2.dll.a -o bzip2.dll libbz2.def $(OBJS)
 
 check: test
 test: bzip2
